From: Christopher Huhn <C.Huhn@gsi.de>
Date: Fri, 23 Nov 2018 17:05:32 +0100
Subject: Sort DHCP entries of each host by its option number

---
 build_isc_dhcp.inc.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build_isc_dhcp.inc.php b/build_isc_dhcp.inc.php
index fceddca..66558dc 100644
--- a/build_isc_dhcp.inc.php
+++ b/build_isc_dhcp.inc.php
@@ -446,6 +446,7 @@ function build_hosts($server_id=0) {
                 dns D,
                 domains Z,
                 interfaces I LEFT OUTER JOIN dhcp_option_entries B ON I.host_id = B.host_id
+                LEFT JOIN dhcp_options opts ON B.dhcp_option_id = opts.id
         WHERE   I.mac_addr NOT like ''
         AND     I.host_id = H.id
         AND     D.domain_id = Z.id
@@ -468,7 +469,7 @@ function build_hosts($server_id=0) {
                                 OR      secondary_server_id = {$server_id}))
                  )
                 )
-        ORDER BY I.ip_addr";
+        ORDER BY I.ip_addr, opts.number";
 
 
 
